client-core/src/service.rs

```diff
@@ -1,20 +1,86 @@
 //! Management services
-mod global_state_service;
-
 mod hd_key_service;
 mod key_service;
 mod multi_sig_session_service;
 mod root_hash_service;
+mod sync_state_service;
 mod wallet_service;
 mod wallet_state_service;
 
 #[doc(hidden)]
 pub use self::wallet_state_service::WalletStateMemento;
 
-pub use self::global_state_service::{GlobalState, GlobalStateService};
 pub use self::hd_key_service::HdKeyService;
 pub use self::key_service::KeyService;
 pub use self::multi_sig_session_service::MultiSigSessionService;
 pub use self::root_hash_service::RootHashService;
-pub use self::wallet_service::WalletService;
-pub use self::wallet_state_service::WalletStateService;
+pub use self::sync_state_service::{SyncState, SyncStateService};
+pub use self::wallet_service::{Wallet, WalletService};
+pub use self::wallet_state_service::{WalletState, WalletStateService};
+
+use secstr::SecUtf8;
+
+use self::sync_state_service::KEYSPACE as SYNC_STATE_KEYSPACE;
+use self::wallet_service::KEYSPACE as WALLET_KEYSPACE;
+use self::wallet_state_service::KEYSPACE as WALLET_STATE_KEYSPACE;
+use client_common::{Result, SecureStorage, Storage};
+
+/// Load wallet from storage
+pub fn load_wallet<S: SecureStorage>(
+    storage: &S,
+    name: &str,
+    passphrase: &SecUtf8,
+) -> Result<Option<Wallet>> {
+    storage.load_secure(WALLET_KEYSPACE, name, passphrase)
+}
+
+/// Save wallet to storage
+pub fn save_wallet<S: SecureStorage>(
+    storage: &S,
+    name: &str,
+    passphrase: &SecUtf8,
+    wallet: &Wallet,
+) -> Result<()> {
+    storage.save_secure(WALLET_KEYSPACE, name, passphrase, wallet)
+}
+
+/// Load wallet state from storage
+pub fn load_wallet_state<S: SecureStorage>(
+    storage: &S,
+    name: &str,
+    passphrase: &SecUtf8,
+) -> Result<Option<WalletState>> {
+    storage.load_secure(WALLET_STATE_KEYSPACE, name, passphrase)
+}
+
+/// Save wallet state to storage
+pub fn save_wallet_state<S: SecureStorage>(
+    storage: &S,
+    name: &str,
+    passphrase: &SecUtf8,
+    state: &WalletState,
+) -> Result<()> {
+    storage.save_secure(WALLET_STATE_KEYSPACE, name, passphrase, state)
+}
+
+/// Load sync state from storage
+pub fn load_sync_state<S: Storage>(storage: &S, name: &str) -> Result<Option<SyncState>> {
+    storage.load(SYNC_STATE_KEYSPACE, name)
+}
+
+/// Save sync state from storage
+pub fn save_sync_state<S: Storage>(storage: &S, name: &str, state: &SyncState) -> Result<()> {
+    storage.save(SYNC_STATE_KEYSPACE, name, state)
+}
+
+/// Delete sync state from storage
+pub fn delete_sync_state<S: Storage>(storage: &S, name: &str) -> Result<()> {
+    storage.delete(SYNC_STATE_KEYSPACE, name)?;
+    Ok(())
+}
+
+/// Delete wallet state from storage
+pub fn delete_wallet_state<S: Storage>(storage: &S, name: &str) -> Result<()> {
+    storage.delete(WALLET_STATE_KEYSPACE, name)?;
+    Ok(())
+}
```

Comment:
The idea of having these functions inside their own service was separation-of-concerns. Having these functions outside of their service goes against the principle of SoC. I'd suggest removing these functions to keep the design more modular.

Reply:
I just thought it might be too many changes a time if I modified too much in the specific service modules, which are depended by others.
I believe in the end, we only need three methods(`load`/`save`/`modify`) for each of `Wallet`/`WalletState`/`SyncState`.

Reply:
I think it's good to move them into different service modules for now, I'll do it.

Reply:
We can make those changes later (or in this PR? Your choice.) if needed. But, this seems to be an unnecessary hack.

Reply:
Done